class-bookrestcontroller.php

```diff
@@ -0,0 +1,213 @@
+			)
+		);
+		//delete item
+		register_rest_route(
+			$this->namespace,
+			//use id and post_id as request parameters since both are needed in the current format
+			'/book/(?P<id>\d+)&(?P<post_id>\d+)',
+			array(
+				'methods'             => 'DELETE',
+				'callback'            => array( $this, 'delete_item' ),
+				'permission_callback' => function () {
+					return $this->cap;
+				},
+			)
+		);
+	}
+	/**
+	 * Retrieves all items from the post meta in JSON format<br>
+	 *
+	 * @param array $request MUST have the key 'id'
+	 *
+	 * @return false|string The data in format JSON or FALSE if bad request
+	 */
+	public function get_items( $request ) {
+		if ( ! $this->validate_int( $request['id'] ) ) {
+			http_response_code(400);
+			return json_encode( array( 'success' => false ) );
+		}
+		$data = get_post_meta( (int) $request['id'], $this->meta_key, true );
+		if ( $data !== false ) {
+			$data = unserialize( $data, array( 'allowed_classes' => true ) );
+		}
+		return json_encode( $data );
+	}
+	/**
+	 * Adds an item into the serialized array<br>
+	 * If the item with the specified id already exists it will be updated
+	 * @param array $request MUST have the following keys : 'id', 'post_id', 'title', 'author' 'genre' and 'summary'
+	 *                       for the operation to be successful
+	 * @return false|string Whether the operation was successful in JSON format
+	 */
+	public function create_item( $request ) {
+		$request->sanitize_params();
+		if ( $this->validate_int( $request['id'] ) || $this->validate_int( $request['post_id'] ) ) {
+			$post_id = (int) $request['post_id'];
+		} else {
+			http_response_code(400);
+			return null;
+		}
+		if ( ! (
+		isset( $request['title'], $request['author'], $request['genre'], $request['summary'] )
+		) ) {
+			http_response_code(400);
+			return null;
+		}
+		if ( ! $this->validate_request_params( $request ) ) {
+			http_response_code(400);
+			return null;
+		}
+		$model = array(
+			'id'      => (int) $request['id'],
+			'title'   => $request['title'],
+			'author'  => $request['author'],
+			'genre'   => $request['genre'],
+			'summary' => $request['summary'],
+		);
+		$data  = unserialize( get_post_meta( $post_id, $this->meta_key, true ), array( 'allowed_classes' => true ) );
+		if ( $data === false ) {
+			$data = array();
+		}
+		$data[ $model['id'] ] = $model;
+		$data                 = serialize( $data );
+		if ( ! update_post_meta( $post_id, $this->meta_key, $data ) ) {
+			http_response_code(400);
+			return null;
+		}
+		return json_encode( array( 'success' => true ) );
+	}
+	/**
+	 * Updates only the specified fields of a book from the serialized array<br>
+	 *
+	 * @param array $request MUST have at least two keys, 'id' and 'post_id'
+	 *
+	 * @return false|string Whether the operation was successful or not as JSON
+	 */
+	public function update_item( $request ) {
+		$request->sanitize_params();
+		//var_dump($request['author']);die; working on allowing names like O'hara
```

Comment:
delete debugging comments

Reply:
> delete debugging comments

here is where I stopped Friday, it shall be removed